### PR TITLE
Many improvements

### PR DESCRIPTION
**Pull request by eugpermar**

### CODE REVIEW

**Comment by eugpermar** [comment]
* Changed rdlog logic (sorry about delay)
* Achieved rd_memctx_t _F_TRACK individual chunk free

**Comment by eugpermar** [comment]
Hi! It's been a long time since this PR with many improvements I need to add to librd in order to work properly in my projects.

I just add a few changes to make it compile on clang-7 default settings. I could cherry-pick only the changes you are interested in (of all of them) and merge them separately (https://github.com/edenhill/librd/pull/12/commits).

Regards!

**Review by edenhill — CHANGES_REQUESTED** [review]

File: rdavl.h [hunk 1]
```diff
@@ -145,7 +145,21 @@ typedef struct rd_avl_s {
  */
 #define RD_AVL_ELM_GET_NL(ran)      ((ran)->ran_elm)
 
+/**
+ * Run a function in every node of the avl tree
+ *
+ * NOTE: can't insert / delete with it, but you can use for free all nodes
+ */
+#define RD_AVL_FOREACH(ran, callback, opaque) \
+	rd_avl_foreach(ravl, callback, opaque, 1);
```
[the remark this answers is not in the document]
Comment (eugpermar) [reply rdavl.h]: ouch! Solved!

**Comment by edenhill** [comment]
Sweet thanks!

**Comment by eugpermar** [comment]
Thank you! :)

**Pull request merged**